src/rgw/CMakeLists.txt

```diff
@@ -419,7 +327,7 @@ target_link_libraries(rgw_schedulers
 set(radosgw_srcs
   rgw_main.cc)
 
-add_executable(radosgw ${radosgw_srcs})
+add_library(radosgw SHARED $<TARGET_OBJECTS:rgw_a> ${radosgw_srcs})
```

Comment:
Because it was thought that some plugins would, or could, have references to things  in the "core" of radosgw and the plugins would need to link with them to resolve those references. (Fedora, and by extension RHEL has a requirement that all shlibs link with the option to not allow undefined refs.) It doesn't seem that the rados plugin references anything though.

Reply:
fixed in the next push